File: example/java/jimmer-sql2/src/main/java/org/babyfish/jimmer/sql2/example/model/common/BaseEntity.java
package org.babyfish.jimmer.sql2.example.model.common;

import org.babyfish.jimmer.pojo.AutoScalarRule;
import org.babyfish.jimmer.pojo.AutoScalarStrategy;
import org.babyfish.jimmer.sql.MappedSuperclass;

import java.time.LocalDateTime;

/*
 * see CommonEntityDraftInterceptor
 */
@MappedSuperclass
@AutoScalarRule(AutoScalarStrategy.NONE)
public interface BaseEntity {

    LocalDateTime createdTime();

    LocalDateTime modifiedTime();
}

File: example/kotlin/jimmer-sql-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/sql/model/common/BaseEntity.kt
package org.babyfish.jimmer.example.kt.sql.model.common

import org.babyfish.jimmer.pojo.AutoScalarRule
import org.babyfish.jimmer.pojo.AutoScalarStrategy
import org.babyfish.jimmer.sql.MappedSuperclass
import java.time.LocalDateTime

/*
 * see CommonEntityDraftInterceptor
 */
@MappedSuperclass
@AutoScalarRule(AutoScalarStrategy.NONE)
interface BaseEntity {

    val createdTime: LocalDateTime

    val modifiedTime: LocalDateTime
}